src/fidesops/graph/data_type.py

```diff
@@ -49,6 +62,10 @@ def empty_value(self) -> int:
         """Empty int value"""
         return 0
 
+    def truncate(self, length: int, val: int) -> int:
+        """Truncates value to given length"""
+        return int(str(val)[:length])
```

Comment:
Probably not the right way to truncate an int. 
The reason for our truncation is so that we don't generate values that will fail on insert when we mask data because they won't fit in the target database. So, e.g. here are mysql types: https://dev.mysql.com/doc/refman/8.0/en/integer-types.html
These are truncated by bytes, so, e.g. if you had a mysql column defined as an INT and you tried to insert a value greater than 2147483647 it would fail.

I think this can be deferred ( for the moment at least - it was never supported in atlas AFAIK) but we shouldn't be truncating alphabetically, in any case.

Reply:
got it. So would byte truncation be diff for nosql vs mysql?

Reply:
I don't really know. The thing I posted was mysql specific.  All of the nosql databases I've worked with sidestep this issue by basically allowing you to write anything - e.g. dynamo _only_ uses BigDecimals in the java implementation . 
I think it shoiuld only matter with defined schema types.